src/adapter/tracer/tests.rs

```diff
@@ -0,0 +1,226 @@
+// The Trustfall API requires the adapter to be passed in as an Arc.
+// Our adapter is not Send/Sync (it doesn't need it),
+// but there's currently nothing we can do about this lint.
+#![allow(clippy::arc_with_non_send_sync)]
```

Comment:
```suggestion
#![expect(clippy::arc_with_non_send_sync)]
```
so that clippy _asserts_ this lint is still here, or otherwise forces us to remove it